sktime/utils/ts_format.py

```diff
@@ -0,0 +1,189 @@
+    ts_df = pd.DataFrame(columns=csv_df_columns)
+    for val in csv_group_by_column_vals:
+        curr_df = csv_df.loc[csv_df[group_by] == val]
+        temp_dict = {}
+        for column in csv_df_columns:
+            temp_dict[column] = curr_df[column]
+        ts_df = ts_df.append(temp_dict, ignore_index=True)
+    return ts_df, csv_group_by_column_vals
+
+
+def write_df_to_data(ts_labels, df, out_path, dimensions):
+    """
+    Writes df to data in .ts format
+
+    Parameters
+    ----------
+    ts_labels: dict
+        Dict containing the header labels for the ts format
+    df: dataframe
+        Dataframe containing the data to be written to .ts format
+    out_path: str
+        String path to output files
+    """
+    file_data = ""
+    for key in ts_labels:
+        file_data += "@" + str(key) + " " + str(ts_labels[key]) + "\n"
+    file_data += "@data"
+    lines = []
+    class_label = []
+    for key in df:
+        i = 0
+        df_dict = df[key].to_dict()
+        for dict_key in df_dict:
+            curr_series = df_dict[dict_key]
+            if type(curr_series) != pd.Series:
+                class_label.append(str(curr_series))
+            else:
+                curr_arr = curr_series.to_list()
+                curr_str_vals = ""
+                for val in curr_arr:
+                    curr_str_vals += str(val)
+                if i >= len(lines):
+                    lines.append(curr_str_vals)
+                else:
+                    lines[i] += ":" + curr_str_vals
+            i += 1
+    for i in range(0, len(lines)):
+        if class_label != []:
+            lines[i] += ":" + class_label[i]
+        file_data += "\n" + lines[i]
+    if ".ts" not in out_path:
+        out_path += ".ts"
+    with open(out_path, "w") as file:
+        file.write(file_data)
+    return file_data
+
+
+def csv_to_ts_format(
+    problem_name,
+    path_to_csv,
+    out_path,
+    group_by,
+    class_label_column=None,
+    timestamps="false",
+):
+    """
+    Loads data from a csv and writes out the ts format of that data. This
+    includes setting correct headers
+
+    Parameters
+    ----------
+    problem_name: str
+        The desired name for the dataset
+    path_to_csv: str
+        The full pathname for the .csv file to read
+    out_path: str
+        The full pathname for the .ts file to be written to
+    group_by: str
+        The name of the column to group by
+    class_label_column: int
+        Integer that is the index of the class label column
+    timestamps: boolean
+        Optional parameter defaulting to False that states if the data has a
+        timestamp
+    return_headers: boolean
+        Optional parameter defaulting to False that returns the headers set
+        (mainly used for testing)
+    """
+    csv_df = pd.read_csv(path_to_csv)
+    df, csv_group_by_column_vals = csv_to_ts_df(csv_df, group_by)
+
+    dimensions = len(df.columns)
+
+    columns = df.columns
```

Comment:
what does code below do? perhaps it's worth it wrapping it in another function for modularity/readability, something like `_infer_ts_headers()`